File: packages/sdk-py/examples/list_assets.py
"""List all assets in the demo portfolio."""
import os
from gecko_vpp_sdk import GeckoVPPClient

with GeckoVPPClient(
    base_url=os.environ.get("GECKO_API", "http://localhost:8000"),
    tenant_id=os.environ.get("GECKO_TENANT", "11111111-1111-1111-1111-111111111111"),
) as client:
    assets = client.assets()
    print(f"Found {len(assets)} assets:")
    for a in assets:
        print(f"  {a['code']:<20} {a['display_name']:<30} {a['asset_class']:<8} {a['capacity_mw']:>7} МВт  {a.get('region') or ''}")

File: packages/sdk-ts/examples/list_assets.ts
import { GeckoVPPClient } from "../src/index.js";

const client = new GeckoVPPClient({
  baseURL: process.env.GECKO_API ?? "http://localhost:8000",
  tenantId: process.env.GECKO_TENANT ?? "11111111-1111-1111-1111-111111111111",
});

const assets = await client.assets.list();
console.log(`Found ${assets.length} assets:`);
for (const a of assets) {
  console.log(`  ${a.code.padEnd(20)} ${a.display_name.padEnd(30)} ${a.asset_class.padEnd(8)} ${a.capacity_mw} МВт  ${a.region ?? ""}`);
}
